src/dmd/dsymbolsem.d

```diff
@@ -3901,61 +3901,64 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
         Linterfaces:
             bool foundVtblMatch = false;
 
-            foreach (b; cd.interfaces)
+            for (ClassDeclaration bcd=cd; foundVtblMatch is false && bcd !is null; bcd=bcd.baseClass)
```

Comment:
please put spaces between the assignments `bcd = cd` and `bcd = bcd.baseClass`

also `!foundVtblMatch  && bcd`